addons/openebs/1.12.0/install.sh

```diff
@@ -256,3 +259,149 @@ function openebs_cstor_pool_count() {
+    fi
+}
+
+function openebs_upgrade_pools() {
+    # NOTE: slightly different arguments to pass for pre and after 1.9.0.
+    # Since we only support 1.6.0 -> 1.12.0 using prefixs for pre 1.9.0.
+    local manifest="/tmp/openebs_pool_job.yaml"
+    local pools=$(kubectl get spc --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}')
+    
+    # Bulk upgrade only supported for versions >=1.9.0.
+    # Have to create a job per pool.
+    for pool in $pools; do
+        local spcCurrentVer=$(kubectl get spc $pool -o jsonpath='{.versionDetails.status.current}')
+        logSubstep "Upgrading $pool from $spcCurrentVer to $OPENEBS_VERSION"
+        local out_file=/tmp/openebs-pool-$pool.yaml
+        cat <<UPGRADE_POOLS >$out_file
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: cstor-spc-$RANDOM
+  namespace: $OPENEBS_NAMESPACE
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      serviceAccountName: openebs-maya-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-spc"
+        - "--from-version=$spcCurrentVer"
+        - "--to-version=$OPENEBS_VERSION"
+        - "--spc-name=$pool"
+        - "--v=4"
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+        image: quay.io/openebs/m-upgrade:$OPENEBS_VERSION
+        imagePullPolicy: Always
+      restartPolicy: OnFailure
+UPGRADE_POOLS
+
+        kubectl apply -f $out_file
+    done
+    logSubstep "OpenEBS batch job(s) to upgrade cStor pools added."
+    # TODO: Validation and user feedback of the successful completion of the jobs.
+    # The jobs migth take a while to run however, blocking at this point isn't necesseraly the best aproach.
+}
+
+function openebs_upgrade_vols() {
+    # NOTE: slightly different arguments to pass for pre and after 1.9.0.
+    # Since we only support 1.6.0 -> 1.12.0 using prefixs for pre 1.9.0.
+    local vols=$(kubectl get pods -l app=cstor-volume-manager -n $OPENEBS_NAMESPACE -o jsonpath='{.items[*].metadata.labels.openebs\.io/persistent-volume}')
+
+    # Bulk upgrade only supported for versions >=1.9.0.
+    # Have to create a job per pv.
+    for pv in $vols; do
+        local out_file=/tmp/openebs-vol-${pv##*-}.yaml
+        cat <<UPGRADE_VOLS >$out_file
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: cstor-vol-$RANDOM
+  namespace: $OPENEBS_NAMESPACE
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      serviceAccountName: openebs-maya-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-volume"
+        - "--from-version=1.6.0"
+        - "--to-version=$OPENEBS_VERSION"
+        - "--pv-name=$pv"
+        - "--v=4"
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+        image: quay.io/openebs/m-upgrade:$OPENEBS_VERSION
+        imagePullPolicy: Always
```

Comment:
here too

Reply:
@areed please review the latest commit addressing this.